src/abstractdataframe/io.jl

```diff
@@ -164,128 +155,143 @@ function html_escape(cell::AbstractString)
     return cell
 end
 
-function _show(io::IO, ::MIME"text/html", df::AbstractDataFrame;
-               summary::Bool=true, eltypes::Bool=true, rowid::Union{Int, Nothing}=nothing)
+function _show(io::IO,
+               ::MIME"text/html",
+               df::AbstractDataFrame;
+               summary::Bool=true,
+               eltypes::Bool=true,
+               rowid::Union{Int, Nothing}=nothing,
+               title::String="",
+               kwargs...)
     _check_consistency(df)
 
-    # we will pass around this buffer to avoid its reallocation in ourstrwidth
-    buffer = IOBuffer(Vector{UInt8}(undef, 80), read=true, write=true)
+    names_str = names(df)
+    types = Any[eltype(c) for c in eachcol(df)]
+    types_str = batch_compacttype(types, 9)
+    types_str_complete = batch_compacttype(types, 256)
 
-    if rowid !== nothing
-        if size(df, 2) == 0
-            rowid = nothing
-        elseif size(df, 1) != 1
-            throw(ArgumentError("rowid may be passed only with a single row data frame"))
+    # For consistency, if `kwargs` has `compact_printng`, we must use it.
```

Comment:
I understand that by default we use compact printing?

Reply:
Yes, if it is not in the IOContext.